src/diffpy/utils/tools.py

```diff
@@ -242,14 +250,15 @@ def compute_mu_using_xraydb(sample_composition, energy, sample_mass_density=None
             "You must specify either sample_mass_density or packing_fraction, but not both. "
             "Please rerun specifying only one."
         )
-    if sample_mass_density is not None:
-        mu = material_mu(sample_composition, energy * 1000, density=sample_mass_density, kind="total") / 10
-    else:
-        warnings.warn(
-            "Warning: Density is set to None if a packing fraction is specified, "
-            "which may cause errors for some materials. "
-            "We recommend specifying sample mass density for now. "
-            "Auto-density calculation is coming soon."
-        )
-        mu = material_mu(sample_composition, energy * 1000, density=None, kind="total") * packing_fraction / 10
+    if packing_fraction is None:
+        packing_fraction = 1
+        try:
+            sample_mass_density = get_density_from_cloud(sample_composition) * packing_fraction
+        except NotImplementedError:
+            warnings.warn(
```

Comment:
I think this needs to be an error not a warning, no?  We need to interrupt execution.

Also, let's apologise... "So sorry, Density computation from composition is not implemented right now.   We hope to have this implemented in the next release. Please rerun specifying a sample mass density.

Reply:
Got it! I will add the COD database and change the error message